asapdiscovery-data/asapdiscovery/data/postera/manifold_artifacts.py

```diff
@@ -29,90 +32,81 @@ class ArtifactType(Enum):
-        s3: S3 = None,
-        manifold_id_column: str = DockingResultCols.LIGAND_ID.value,
-    ):
-        """
-        Parameters
-        ----------
-        molecule_dataframe : pd.DataFrame
-            The dataframe containing the molecules to upload.
-            Must contain a column with the name of the artifact
-        molecule_set_id : UUID
-            The UUID of the molecule set to upload to
-        artifact_type : ArtifactType
-            The type of artifact to upload
-        moleculeset_api : MoleculeSetAPI
-            The MoleculeSetAPI object to use to upload to Manifold
-        cloudfront : CloudFront
-            The CloudFront object to use to generate signed urls
-        s3 : S3
-            The S3 object to use to upload to S3
-        target : str
-            The biological target string for the artifact, one of asapdiscovery.data.postera.manifold_data_validation.TargetTags
-        artifact_column : str
-            The name of the column containing the filesystem path to the artifact that will be uploaded.
-        bucket_name : str
-            The name of the bucket to upload to
-        manifold_id_column : str
-            The name of the column containing the manifold id
-        """
-        self.target = target
-        self.molecule_dataframe = molecule_dataframe.copy()
-        self.molecule_set_id_or_name = molecule_set_id_or_name
-        self.bucket_name = bucket_name
-        self.artifact_columns = artifact_columns
-        self.artifact_types = artifact_types
-
-        if moleculeset_api is None:
-            moleculeset_api = MoleculeSetAPI().from_settings(PosteraSettings())
-        self.moleculeset_api = moleculeset_api
-
-        self.molset_id = self.moleculeset_api.molecule_set_id_or_name(
-            self.molecule_set_id_or_name, self.moleculeset_api.list_available()
-        )
-
-        if cloudfront is None:
-            cloudfront = CloudFront.from_settings(CloudfrontSettings())
-        self.cloudfront = cloudfront
-
-        if s3 is None:
-            s3 = S3.from_settings(S3Settings())
-        self.s3 = s3
-
-        self.manifold_id_column = manifold_id_column
-        if len(self.artifact_columns) != len(self.artifact_types):
+logger = logging.getLogger(__name__)
+
+
+class ManifoldArtifactUploader(BaseModel):
+    target: TargetTags = Field(
+        None, description="The biological target string for the artifact"
+    )
+    molecule_dataframe: pd.DataFrame = Field(
+        ...,
+        description="The dataframe containing the molecules and artifacts to upload",
+    )
+    molecule_set_id: Optional[str] = Field(
+        None, description="The UUID of the molecule set to upload to"
+    )
+    molecule_set_name: Optional[str] = Field(
+        None, description="The name of the molecule set to upload to"
+    )
+
+    bucket_name: str = Field(..., description="The name of the bucket to upload to")
+
+    artifact_columns: list[str] = Field(
```

Comment:
Should the type include optional if the default is None?

Reply:
A couple of the other fields also seem to have None as the default value but its not a expected type.

Reply:
Those are designed to throw if nothing is passed, ie there is no sensible default value it must be provided. Is there a better way to acheive that?

Reply:
Ah right so would `...` work?

Reply:
No, it should throw I think as will fail the validation.